supervisor/host/network.py

```diff
@@ -345,28 +347,38 @@ class Interface:
     @staticmethod
     def from_dbus_interface(inet: NetworkInterface) -> Interface:
         """Concert a dbus interface into normal Interface."""
+        ipv4_method = (
+            Interface._map_nm_method(inet.settings.ipv4.method)
+            if inet.settings and inet.settings.ipv4
+            else InterfaceMethod.DISABLED
+        )
+        ipv6_method = (
+            Interface._map_nm_method(inet.settings.ipv6.method)
+            if inet.settings and inet.settings.ipv6
+            else InterfaceMethod.DISABLED
+        )
         return Interface(
             inet.name,
             inet.settings is not None,
             Interface._map_nm_connected(inet.connection),
             inet.primary,
             Interface._map_nm_type(inet.type),
             IpConfig(
-                Interface._map_nm_method(inet.settings.ipv4.method),
+                ipv4_method,
                 inet.connection.ipv4.address,
                 inet.connection.ipv4.gateway,
                 inet.connection.ipv4.nameservers,
```

Comment:
Can't we get those 3 values from the settings object as well? Then the if statement below needs changing too, it would make the separate checking above unnecessary and simplify the code.

Reply:
I see, this would really make `Interface` class not having the current IP information, which would affect our APIs etc.

IMHO, class `Interface` should have a *current* IPv4/IPv6 field as well as a *configured* IPv4/IPv6 field. This is how NM works under the hood, and would avoid these type of problems we are seeing.

Reply:
So to my knowledge there's only one possible issue this could cause. If it's somehow possible for an interface using `method: static` to have different values in active connection info then in its settings. Because here is how this information is actually used to make the update call to NetworkManager:
https://github.com/home-assistant/supervisor/blob/5919bc22526832d84b8cdb6596e57f9ce60535f3/supervisor/dbus/network/setting/generate.py#L63-L113

As you can see, we don't even look at those other fields unless the method is static. If we think this is a risk I can account for it in the code but I don't see how. We can't just change this though because the response here is returned from the API and the user expects to see the current address, gateway and nameserver info when they use `auto`

Reply:
And when I say account for it I mean here:
https://github.com/home-assistant/supervisor/blob/5919bc22526832d84b8cdb6596e57f9ce60535f3/supervisor/host/network.py#L105-L126

This is the only spot where that can be an issue. In this code after calling `from_dbus_interface` I would replace the ipv4 and ipv6 `addresses`, `gateway` and `nameserver` values with what is in settings if the method is static. Because that is the only time the response of `from_dbus_interface` is passed directly to an update call.

The other time `from_dbus_interface` is used in an update call is indirectly. The user is presented the output of `from_dbus_interface` from the API in the UI. They then can choose to submit what they are seeing to update the network interface. At that point it really doesn't matter what's in settings, the user has just told us what they want the settings to be.